bug: does not handle terraform optional attributes in object types

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->

Given the following valid Terraform code tfsec will panic (not a string).

````
provider "azurerm" {
   features {}
}

variable "net_rule" {
    type = object({
        default_action = optional(string, "Deny")
    })
    default = {}
}

resource "azurerm_storage_account" "storageaccount2" {
  name                             = "dummy"
  resource_group_name              = "dummy"
  location                         = "uksouth"
  account_replication_type         = "LRS"
  account_tier                     = "Premium"

  network_rules {
    default_action = var.net_rule.default_action
    }
  }

````

**To Reproduce**
Steps to reproduce the behavior:
1. Place the above code in a fine in an empty directory
2. Reun terraform init
3. Run terraform plan (to demonstrate valid code)
4. Run tfsec

**Expected behavior**
tfsec should report a number of security issues, eg 2 passed and 2 potential problems detected.
**Output of your tfsec command with --debug flag**
````
00:26.418808600 system.info                      APP       tfsec
00:26.418940100 system.info                      VERSION   v1.28.0
00:26.418947100 system.info                      OS        linux
00:26.418954500 system.info                      ARCH      amd64
00:26.418959800 system.info                      KERNEL    Linux version 5.10.102.1-microsoft-standard-WSL2 (oe-user@oe-host) (x86_64-msft-linux-gcc (GCC) 9.3.0, GNU ld (GNU Binutils) 2.34.0.20200220) #1 SMP Wed Mar 2 00:30:59 UTC 2022
00:26.418967900 system.info                      TERM      xterm-256color
00:26.418972400 system.info                      SHELL     /bin/bash
00:26.418976700 system.info                      GOVERSION go1.18.5
00:26.418981500 system.info                      GOROOT    /opt/hostedtoolcache/go/1.18.5/x64
00:26.418987100 system.info                      CGO       false
00:26.418992700 system.info                      CPUCOUNT  4
00:26.418997100 system.info                      MAXPROCS  4
00:26.419001500 system.info                      WORKDIR   /home/noel-jones/reps/tfsec_test2
00:26.419007000 system.info                      UID       1000
00:26.419021200 system.info                      EUID      1000
00:26.419029500 system.info                      DOCKER    false
00:26.419092500 system.info                      CI        false
00:26.419099300 system.info                      HOSTNAME  UKTECTXVDI23P
00:26.419111200 system.info                      TEMP      /tmp
00:26.419191500 system.info                      PATHSEP   /
00:26.419352000 system.info                      CMD       tfsec --debug
00:26.419495600 cmd                              Command args=[]string{}
00:26.419517700 cmd                              Determined path dir=/home/noel-jones/reps/tfsec_test2
00:26.419567800 cmd                              Determined path root=/
00:26.419573200 cmd                              Determined path rel=home/noel-jones/reps/tfsec_test2
00:26.419832900 terraform.scanner                Scanning [&{/ /}] at 'home/noel-jones/reps/tfsec_test2'...
00:26.421552800 terraform.scanner.rego           Loaded 4 embedded libraries.
00:26.490689600 terraform.scanner.rego           Loaded 117 embedded policies.
00:26.677158300 terraform.scanner                Scanning root module 'home/noel-jones/reps/tfsec_test2'...
00:26.677323100 terraform.parser.<root>          Setting project/module root to 'home/noel-jones/reps/tfsec_test2'
00:26.677402500 terraform.parser.<root>          Parsing FS from 'home/noel-jones/reps/tfsec_test2'
00:26.677479200 terraform.parser.<root>          Parsing 'home/noel-jones/reps/tfsec_test2/no_default_action.tf'...
00:26.678015900 terraform.parser.<root>          Added file home/noel-jones/reps/tfsec_test2/no_default_action.tf.
00:26.678228800 terraform.parser.<root>          Evaluating module...
00:26.679495800 terraform.parser.<root>          Read 3 block(s) and 0 ignore(s) for module 'root' (1 file[s])...
00:26.679804400 terraform.parser.<root>          Added 0 variables from tfvars.
00:26.679953800 terraform.parser.<root>          Error loading module metadata: open //home/noel-jones/reps/tfsec_test2/.terraform/modules/modules.json: no such file or directory.
00:26.680080700 terraform.parser.<root>          Working directory for module evaluation is '/home/noel-jones/reps/tfsec_test2'
00:26.681696400 terraform.parser.<root>.evaluator Filesystem key is 'f9088d3ed2db9899500f703a07bb505300c2b5cbc122ac4365ca04af35422e64'
00:26.681795000 terraform.parser.<root>.evaluator Starting module evaluation...
00:26.682163400 terraform.parser.<root>.evaluator Starting submodule evaluation...
00:26.682287500 terraform.parser.<root>.evaluator Finished processing 0 submodule(s).
00:26.683489700 terraform.parser.<root>.evaluator Starting post-submodule evaluation...
00:26.684259500 terraform.parser.<root>.evaluator Module evaluation complete.
00:26.684396900 terraform.parser.<root>          Finished parsing module 'root'.
00:26.685190700 terraform.executor               Adapting modules...
panic: not a string

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.AsString({{{0x0?, 0x0?}}, {0x0?, 0x0?}})
        /home/runner/go/pkg/mod/github.com/zclconf/go-cty@v1.10.0/cty/value_ops.go:1256 +0x138
github.com/aquasecurity/defsec/internal/adapters/terraform/azure/storage.adaptNetworkRule(0xc000190d80)
        /home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.76.0/internal/adapters/terraform/azure/storage/adapt.go:160 +0x9c
github.com/aquasecurity/defsec/internal/adapters/terraform/azure/storage.adaptAccount(0xc000190cc0)
        /home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.76.0/internal/adapters/terraform/azure/storage/adapt.go:117 +0x4b7
github.com/aquasecurity/defsec/internal/adapters/terraform/azure/storage.adaptAccounts({0xc000937420, 0x1, 0x0?})
        /home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.76.0/internal/adapters/terraform/azure/storage/adapt.go:77 +0x46d
github.com/aquasecurity/defsec/internal/adapters/terraform/azure/storage.Adapt({0xc000937420, 0x1, 0x1})
        /home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.76.0/internal/adapters/terraform/azure/storage/adapt.go:13 +0x46
github.com/aquasecurity/defsec/internal/adapters/terraform/azure.Adapt({_, _, _})
        /home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.76.0/internal/adapters/terraform/azure/adapt.go:34 +0x425
github.com/aquasecurity/defsec/internal/adapters/terraform.Adapt({0xc000937420, 0x1, 0x1})
        /home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.76.0/internal/adapters/terraform/adapt.go:20 +0x7e
github.com/aquasecurity/defsec/pkg/scanners/terraform/executor.(*Executor).Execute(0xc000e43000, {0xc000937420?, 0x1, 0x1})
        /home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.76.0/pkg/scanners/terraform/executor/executor.go:96 +0x11a
github.com/aquasecurity/defsec/pkg/scanners/terraform.(*Scanner).ScanFSWithMetrics(0xc00023a1e0, {0x1e63900, 0xc00012e028}, {0x7fa83cdcd338?, 0xc000d18480?}, {0xc000044005, 0x20})
        /home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.76.0/pkg/scanners/terraform/scanner.go:210 +0x75a
github.com/aquasecurity/tfsec/internal/app/tfsec/cmd.Root.func1(0xc000314a00, {0xc0006d8cd0, 0x0, 0x1})
        /home/runner/work/tfsec/tfsec/internal/app/tfsec/cmd/root.go:83 +0x633
github.com/spf13/cobra.(*Command).execute(0xc000314a00, {0xc00012a010, 0x1, 0x1})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc000314a00)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
        /home/runner/work/tfsec/tfsec/cmd/tfsec/main.go:12 +0x25
````

<!-- If applicable, add screenshots/output to help explain your problem. Remove sensitive info first! -->

**System Info**
<!-- complete the following -->
 - tfsec version: `v1.28.0`
 - terraform version: `1.3.3`
 - OS: `/home/noel-jones/go/bin/tfsec`

**Example Code**

The issue can be simplified to the following code with the same result

````
provider "azurerm" {
   features {}
}

resource "azurerm_storage_account" "storageaccount1" {
  name                             = "dummy"
  resource_group_name              = "dummy"
  location                         = "uksouth"
  account_replication_type         = "LRS"
  account_tier                     = "Premium"

  network_rules {
    default_action = null
    }
  }
````
<!-- a minimal terraform example which will help us reproduce the issue (remove anything sensitive first!) -->
**Additional context**
<!-- Add any other context about the problem here. -->

This is in some respects 2 issues so I am not sure of the best resolution. While the simpler example does not terraform plan the frst does. I imagine that changing tfsec to handle optional attributes within types is a difficult fix, an enhancement perhaps.  For this issue perhaps focus on the fact that it panics which took me a long time to identify the root cause.

--- a/internal/adapters/terraform/azure/storage/adapt.go
+++ b/internal/adapters/terraform/azure/storage/adapt.go
@@ -1,8 +1,6 @@
 package storage
 
 import (
-	"strings"
-
 	defsecTypes "github.com/aquasecurity/defsec/pkg/types"
 
 	"github.com/aquasecurity/defsec/pkg/providers/azure/storage"
@@ -157,12 +155,7 @@ func adaptNetworkRule(resource *terraform.Block) storage.NetworkRule {
 	defaultActionAttr := resource.GetAttribute("default_action")
 
 	if defaultActionAttr.IsNotNil() {
-		switch strings.ToLower(defaultActionAttr.Value().AsString()) {
-		case "allow":
-			allowByDefault = defsecTypes.Bool(true, defaultActionAttr.GetMetadata())
-		case "deny":
-			allowByDefault = defsecTypes.Bool(false, defaultActionAttr.GetMetadata())
-		}
+		allowByDefault = defsecTypes.Bool(defaultActionAttr.Equals("Allow", terraform.IgnoreCase), defaultActionAttr.GetMetadata())
 	} else {
 		allowByDefault = defsecTypes.BoolDefault(false, resource.GetMetadata())
 	}

